File: codility/src/main/java/codility/com/lessons/l07stackandqueue/Brackets.java
package codility.com.lessons.l07stackandqueue;

import java.util.Stack;

/**
 * 100%
 * https://app.codility.com/programmers/lessons/7-stacks_and_queues/brackets/
 */
public class Brackets {
	public int solution(String S) {
		Stack<Character> pilha = new Stack<>();
		char[] charArray = S.toCharArray();
		for (char x : charArray) {
			if (x == '(' || x == '{' || x == '[') {
				pilha.push(x);
			} else {
				if (pilha.isEmpty()) {
					return 0;
				}
				switch (pilha.pop().charValue()) {
					case '(':
						if (x != ')') {
							return 0;
						}
						break;
					case '[':
						if (x != ']') {
							return 0;
						}
						break;
					case '{':
						if (x != '}') {
							return 0;
						}
						break;
				}
			}
		}
		return pilha.isEmpty() ? 1 : 0;
	}
}

File: codility_new/brackets.py
# https://app.codility.com/programmers/lessons/7-stacks_and_queues/brackets/
def solution(S):
    checking = []

    for letter in S:
        if letter == '[' or letter == '{' or letter == '(':
            checking.append(letter)
        else:
            if len(checking) == 0:
                return 0
            poped = checking.pop()
            if poped == '[' and letter != ']':
                return 0
            if poped == '{' and letter != '}':
                return 0
            if poped == '(' and letter != ')':
                return 0

    return 1 if len(checking) == 0 else 0
